### PR TITLE
C# server streaming

### PR DESCRIPTION
**Pull request by chrisdunelm**

### CODE REVIEW

File: src/main/resources/com/google/api/codegen/csharp/gapic_client.snip [hunk 1]
```diff
@@ -526,6 +546,52 @@
...
+            return new {@method.name}StreamImpl({@method.callableName}.Call(request, callSettings));
+        }
+
+        internal sealed class {@method.name}StreamImpl : {@method.name}Stream
+        {
+            /// <summary>
+            /// Construct the server-streaming method for <c>{@method.name}</c>.
+            /// </summary>
+            /// <param name="call">The underlying gRPC server-streaming call.</param>
+            public {@method.name}StreamImpl({@method.responseTypeName} call)
```
Comment (jskeet) [inline src/main/resources/com/google/api/codegen/csharp/gapic_client.snip]: Make the constructor internal as well, just for clarity?

File: src/main/resources/com/google/api/codegen/csharp/gapic_snippets.snip [hunk 2]
```diff
@@ -226,6 +228,30 @@
...
 # snippetMethod: StaticLangApiMethodSnippetView
+@private streamingRequestObjectMethodServer(snippetMethod)
+    public async Task {@snippetMethod.snippetMethodName}()
+    {
+        // Snippet: {@snippetMethod.method.name}({@snippetMethod.method.serviceRequestTypeName},CallSettings)
+        // Create client
+        {@snippetMethod.apiClassName} {@snippetMethod.apiVariableName} = {@snippetMethod.apiClassName}.Create();
+        // Initialize request argument
+        {@topLevelInitCode(snippetMethod.method.initCode)}
+        // Make the request, returning a streaming response
```
Comment (jskeet) [inline src/main/resources/com/google/api/codegen/csharp/gapic_snippets.snip]: The terminology and variable names here are a bit confusing, but I suspect that's inherent. I haven't come up with anything better.

File: src/main/resources/com/google/api/codegen/csharp/gapic_client.snip [hunk 3]
```diff
@@ -526,6 +546,52 @@
 @end
 
 # api: StaticLangApiView, method: StaticLangApiMethodView, isImp: boolean
+@private streamingRequestObjectMethodServer(api, method, isImpl)
+    @if isImpl
+        public override {@method.name}Stream {@method.name}(
```
Comment (jskeet) [inline src/main/resources/com/google/api/codegen/csharp/gapic_client.snip]: Do we not need an async method here as well? Unlike the bidi streaming, I'd expect this to actually make the request.

Comment (chrisdunelm) [inline src/main/resources/com/google/api/codegen/csharp/gapic_client.snip]: The underlying call here is actually asynchronous. I agree this is slightly unexpected.
The return type of the gRPC call is AsyncServerStreamingCall, and it performs an async request.

File: src/main/resources/com/google/api/codegen/csharp/gapic_client.snip (hunk 1 above)
Comment (chrisdunelm) [inline src/main/resources/com/google/api/codegen/csharp/gapic_client.snip]: Done

File: src/main/resources/com/google/api/codegen/csharp/gapic_snippets.snip (hunk 2 above)
Comment (chrisdunelm) [inline src/main/resources/com/google/api/codegen/csharp/gapic_snippets.snip]: ack

**Comment by jskeet** [comment]
Will need to get my head back into this - so will wait until I'm back home, or maybe look tomorrow (so we can discuss in person).

**Comment by chrisdunelm** [comment]
Tomorrow sgtm

File: src/main/resources/com/google/api/codegen/csharp/gapic_client.snip [hunk 4]
```diff
@@ -525,6 +545,52 @@
...
+            internal {@method.name}StreamImpl({@method.responseTypeName} call)
+            {
+                GrpcCall = call;
+            }
+
+            /// <inheritdoc/>
+            public override {@method.responseTypeName} GrpcCall { get; }
+
+            /// <inheritdoc/>
+            public override IAsyncEnumerator<{@method.serviceResponseTypeName}> ResponseStream =>
```
Comment (jskeet) [inline src/main/resources/com/google/api/codegen/csharp/gapic_client.snip]: Collapse to one line?

Comment (chrisdunelm) [inline src/main/resources/com/google/api/codegen/csharp/gapic_client.snip]: Done

**Pull request merged**